Backbase/variants
# Fix GH Action

### What does this PR do
- Fixes our GH action to switch variants, so that it automatically installs the right version using homebrew.
- If using a `macos-12` runner, homebrew will install using a packaged bottle, taking seconds.
- Remove steps to prepare a ci build from a previously committed binary, reliant on a manual step

### How can it be tested
0. Ensure your workflow is running on `macos-12` and not `macos-latest`.

1. Given the following step, variants switch should **succeed**, installing the latest version (since 1.1.4 is higher and inexistent as of now):
```yaml
    - uses: backbase/variants@fix/gh-action
      with:
        version: 1.1.4
        platform: ios
        variant: beta
        verbose: true
```

2. Given the following step, variants switch should **succeed**, installing the latest version (when omitted, 'version' defaults to 'latest'):
```yaml
    - uses: backbase/variants@fix/gh-action
      with:
        platform: ios
        variant: beta
        verbose: true
```

3. Given the following step, variants switch should **succeed**, installing version 1.1.1:
```yaml
    - uses: backbase/variants@fix/gh-action
      with:
        version: 1.1.1
        platform: ios
        variant: beta
        verbose: true
```

4. Given the following step, variants switch should **fail**. The version is expected in the format 'major.minor.patch' / 'x.y.z':
```yaml
    - uses: backbase/variants@fix/gh-action
      with:
        version: 1.1
        platform: ios
        variant: beta
        verbose: true
```

### Task
resolves #189 

### Checklist:
